src/main/java/org/gridsuite/shortcircuit/server/ShortCircuitController.java

```diff
@@ -103,8 +104,9 @@ public ResponseEntity<ShortCircuitAnalysisResult> getResult(@Parameter(descripti
         @ApiResponse(responseCode = "404", description = "Short circuit analysis result has not been found")})
     public ResponseEntity<Page<FaultResult>> getPagedFaultResults(@Parameter(description = "Result UUID") @PathVariable("resultUuid") UUID resultUuid,
                                                                 @Parameter(description = "Full or only those with limit violations or none fault results") @RequestParam(name = "mode", required = false, defaultValue = "WITH_LIMIT_VIOLATIONS") FaultResultsMode mode,
+                                                                @Parameter(description = "Fetch all results or paged results") @RequestParam(name = "allResults", required = false, defaultValue = "false") boolean allResults,
                                                                 Pageable pageable) {
-        Page<FaultResult> faultResultsPage = shortCircuitService.getFaultResultsPage(resultUuid, mode, pageable);
+        Page<FaultResult> faultResultsPage = shortCircuitService.getFaultResultsPage(resultUuid, mode, allResults ? PageRequest.of(0, Integer.MAX_VALUE, pageable.getSort()) : pageable);
```

Comment:
Maybe add check on pageable.getSort() : it should not be null
if it's null, use Sort.unsorted()